sympy/solvers/ode.py

```diff
@@ -5930,7 +5898,24 @@ def _remove_coefficient(expr, x):
         # then they will need to be multiplied by sufficient x to make them so.
         # This function DOES NOT do that (it doesn't even look at the
         # homogeneous equation).
-        retdict['trialset'] = _get_trial_set(expr, x)
+        temp_set = set([])
+        cartesian_prod = set([])
+        if expr.is_Add:
+            for i in expr.args:
```

Comment:
Rather than checking if it's an `Add` just force it to be one:
```
In [26]: Add.make_args(x**2 + exp(x))                                                                    
Out[26]: 
⎛ 2   x⎞
⎝x , ℯ ⎠

In [27]: Add.make_args(x**2)                                                                             
Out[27]: 
⎛ 2 ⎞
⎝x ,⎠
```
Then we don't need two branches.

Reply:
so instead of `if stattement` i should directly write `for i in Add.make_args(expr)`?

Reply:
Actually the goal of making if else statement was : for example if the non - homogeneous part = `sin(x) + x` now lets say only `sin(x)` is the part of complimentary solutions so only family of `sin(x)` should be multiplied by x but the `_get_trial_set(expr)` is written in such a manner that it always gives family of whole expression i.e giving union in this case. but here we need to separate these two families. while for example if non-homogeneous contains `x*sin(x)` then we dont need to check for each family because at the end cartesian product is done